src/gax/src/options.rs

```diff
@@ -188,6 +210,12 @@ pub trait RequestOptionsBuilder: internal::RequestBuilder {
 
     /// Sets the polling backoff policy configuration.
     fn with_polling_backoff_policy<V: Into<PollingBackoffPolicyArg>>(self, v: V) -> Self;
+
+    /// Sets the path template for the request URL.
+    fn with_path_template(self, v: Option<String>) -> Self;
+
+    /// Sets the prior attempt count for the request.
+    fn with_prior_attempt_count(self, v: u32) -> Self;
```

Comment:
You can make this change non-breaking using:


```suggestion
    /// Sets the path template for the request URL.
    fn with_path_template(self, _v: Option<String>) -> Self { self }

    /// Sets the prior attempt count for the request.
    fn with_prior_attempt_count(self, _v: u32) -> Self { self }
```

But we should consider if these methods belong in the public API exposed to customers (I recall methods added here  automatically become part of the public API). I think they should not be public.  This is not an attribute of the request applications should be able to set or would want to set.

Reply:
I'd prefer they are not in the public API -- I can follow the example for default_idempotency where accessors are in the ::internal module.